libdnf/hy-goal.h

```diff
@@ -117,6 +117,16 @@ int hy_goal_install(HyGoal goal, DnfPackage *new_pkg);
 */
 int hy_goal_install_optional(HyGoal goal, DnfPackage *new_pkg);
 
+/**
+* @brief Lock package state. If installed, remains installed. If uninstalled,
+* remains uninstalled. Return value is always 0.
```

Comment:
Assuming this is just for future-proofing the API, wouldn't it be more meaningful to document how errors will be signaled?
That is, if a consumer wants to be future-proof, should it be checking for `< 0`, `> 0` or `!= 0`?

Reply:
Cau you please also add `GError **error` as the last argument, even if it will not be used? While seemingly nothing can go wrong here, there's no way to return any error in the future in case it somehow would be needed. (and it's quite a big problem with some of the API functions that lack this argument)

Reply:
I'm clearly in favour of better error-handling. :) I wrote it this way because the majority of functions it sits alongside have the same signature and return semantics. Totally fine with having an explicit `GError` if we don't mind breaking that mold.

Reply:
Fixed this now!